Test/Utils/ComputeNode/Initialize.ps1

```diff
@@ -26,3 +59,44 @@ function Initialize-ComputeNode {
         Write-Log "Created network id: $ID"
     }
 }
+
+function Install-ComputeServices {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [SystemConfig] $SystemConfig,
+        [Parameter(Mandatory = $true)] [OpenStackConfig] $OpenStackConfig,
+        [Parameter(Mandatory = $true)] [ControllerConfig] $ControllerConfig
+    )
+    Install-Components -Session $Session
+
+    Initialize-ComputeServices `
+        -Session $Session `
+        -SystemConfig $SystemConfig `
+        -OpenStackConfig $OpenStackConfig `
+        -ControllerConfig $ControllerConfig
+}
+
+function Initialize-ComputeServices {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [SystemConfig] $SystemConfig,
+        [Parameter(Mandatory = $true)] [OpenStackConfig] $OpenStackConfig,
+        [Parameter(Mandatory = $true)] [ControllerConfig] $ControllerConfig
```

Comment:
1. 

We have this triple `SystemConfig`, `OpenStackConfig` and `ControllerConfig` passed here all the way from `Initialize-ComputeNode`. Why not just have 
```
[Parameter(Mandatory=$true)] [TestenvConfigs] $Configs
```
here, in `Initialize-ComputeServices`?

2. 

This whole chain of calls Initialize-ComputeNode -> Install-ComputeServices -> Initialize-ComputeServices..

I feel like it should be Initialize-ComputeNode -> Install-ComputeServices && Initialize-ComputeServices

Maybe even something like:

```
function Initialize-ComputeNode {
    Install-ComputeServices
    Set-ComputeServicesConfiguration
    Start-ComputeServices
}

function Install-ComputeServices {
    Install-Components
}

function Set-ComputeServicesConfiguration {
    New-NodeMgrConfigFile
    New-CNMPluginConfigFile
    New-AgentConfigFile
}

function Start-ComputeServices {
    Initialize-DriverAndExtension
    Start-AgentService
}
```

This way we don't create this scary chain of function calls and everything is kinda flat.

3. 

Also, maybe a rename Initialize-ComputeNode -> Initialize-FullWindowsComputeNode

Name changes are really nitpicky and I hate suggesting them and I'm sorry. But I'm always super scared about our init/cleanup code in these tests.

Reply:
2.

you're right https://github.com/Juniper/contrail-windows-ci/blob/6f5171fc82c6a21b66871a04d1ecc0f723dfdfd9/Test/Utils/ComputeNode/Configuration.ps1#L167

meh... I still think having something like the function below is better than 3-level chain, although it's less elegant

```
function Initialize-ComputeNode {
    Install-ComputeServices

    New-NodeMgrConfigFile
    New-CNMPluginConfigFile
    Start-DriverAndExtension

    New-AgentConfigFile
    Start-AgentService
}
```